### PR TITLE
Add library building

### PR DESCRIPTION
**Pull request by trevordmiller**
# Changes

- Fix a few errors that were happening when viewing the style guide
- Add `yarn dev:library` script for watching the library build with `yarn link`
- Fix a few errors while testing importing in another project
- Testing by live editing while importing in `egghead-instructor-center` with yarn link
- Add README with complete information
- Remove root level copies that are in `/src` (where misplaced during copy to new repo)
- Remove unused `dependencies` in package.json
- Move all `devDependencies` to `dependencies` for known Heroku issue

# Next Steps

- Review this PR and updated README to ensure everything makes sense
- Merge this into `egghead/egghead-ui`
- Version and publish (`yarn version && git push && git push --tags && npm publish`)

_At this point other projects can start using `egghead-ui`; I'll do this in the instructor center_

- Fix other errors, there are a few; be sure to check console while developing ;)
- Fix mobile issues with components
- Update the missing components to work with the build (Logo, Card); will need some sort of CSS solution; if .css/.svg etc. files are needed, a webpack compilation step will need to be added for the library build.
- Wire up Codeship to do the versioning and publishing when master is updated, so when master is updated both the library (on npm) and style guide (styleguide.egghead.io) are updated automatically after the builds pass.

# Tips

The way I was able to build this and test it is by using [yarn link](https://yarnpkg.com/en/docs/cli/link). This should always be used before publishing to consume the package and make sure it works so when you publish the package has already been pre-tested in a real environment. The way you do this is:

In one terminal pane:
```
$ cd egghead-ui
$ yarn link
$ yarn dev:library
```

In another terminal pane:
```
$ cd some-consuming-project
$ yarn link egghead-ui

# In file in editor

import {SomeComponent} from 'egghead-ui'

<SomeComponent ...
```

Then, when everything is working as expected, you can publish a new version of the library:
```
$ yarn version && git push && git push --tags && npm publish
```
Or use a package like `np` to automate this.

Then, you can use the library in other projects for reals:
```
$ cd egghead-ui
$ yarn unlink
$ cd some-project
$ yarn add egghead-ui

import {SomeComponent} from 'egghead-ui'

<SomeComponent ...
```

# Result For User

It needs work but its working!

screen shot 2017-01-26 at 9 27 33 pm

---

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,5 +4,5 @@ node_modules
 npm-debug.log
 /.idea
 .tern-port
-lib
 build
+lib
```
Comment (tayiorbeii) [inline .gitignore:8]: Does the order matter in a `gitignore`?

File: src/components/Button/index.examples.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ storiesOf('Button')
       <div className={btnDisp}>
         {standardButtonExamples.map((btn, i) => {
           return (
-            <Button type={btn.type} onClick={clickHandlerFixture}>
+            <Button type={btn.type} onClick={clickHandlerFixture} key={i}>
```
Comment (tayiorbeii) [inline src/components/Button/index.examples.js:28]: Can't believe I missed `key`, I'm usually the one catching that for other people! :)

**Review by tayiorbeii — APPROVED** [review]

File: src/components/Button/index.examples.js (hunk 2 above)
Comment (trevordmiller) [reply src/components/Button/index.examples.js:28]: :)

File: .gitignore (hunk 1 above)
Comment (trevordmiller) [reply .gitignore:8]: Nah, I'm just particular and everything else was styleguide then library, nbd though

**Pull request merged**